### PR TITLE
Airport Ticket Booking Unit Tests

### PR DESCRIPTION
**Pull request by mohammed-switi**

### CODE REVIEW

File: Airport_Ticket_Booking_System_Test/BookingServiceTest.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
+// File: Airport_Ticket_Booking_System_Test/BookingServiceTests.cs
```
Comment (R-abodyak) [inline Airport_Ticket_Booking_System_Test/BookingServiceTest.cs]: you can remove

File: Airport_Ticket_Booking_System_Test/BookingServiceTest.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        private readonly IBookingService _bookingService;
+
+        public BookingServiceTests()
+        {
+            _bookingService = new BookingService();
+        }
+
+        [Fact]
+        public void BookFlight_ShouldAddBooking()
```
Comment (R-abodyak) [inline Airport_Ticket_Booking_System_Test/BookingServiceTest.cs]: try to cover all branches in a method 
try adding test for duplicate key case

File: Airport_Ticket_Booking_System_Test/BookingServiceTest.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        {
+            // Create a flight with all required properties.
+            var flight = new Flight("FL001", "USA", "Canada", "JFK", "YYZ", DateTime.UtcNow, new Dictionary<string, decimal> { { "Economy", 500M } });
+            _bookingService.BookFlight(flight.FlightNumber, flight, "Economy");
+            var bookings = _bookingService.GetAllBookings();
+
+            Assert.NotEmpty(bookings);
+            Assert.Contains(bookings, b => b.FlightNumber == "FL001" && b.ClassType == "Economy");
+        }
```
Comment (R-abodyak) [inline Airport_Ticket_Booking_System_Test/BookingServiceTest.cs]: Add 
//Arrange 
//Act 
//Assert

File: Airport_Ticket_Booking_System_Test/BookingServiceTest.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+     
+
+        [Fact]
```
Comment (R-abodyak) [inline Airport_Ticket_Booking_System_Test/BookingServiceTest.cs]: one line between methods is enough

File: Airport_Ticket_Booking_System_Test/BookingServiceTest.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+using System.Linq;
+using Xunit;
+using Airport_Ticket_Booking_System.Models;
+using Airport_Ticket_Booking_System.Interfaces;
+using Airport_Ticket_Booking_System.Services;
+using System.Collections.Generic;
+
+namespace Airport_Ticket_Booking_System_Test
+{
+    public class BookingServiceTests
```
Comment (R-abodyak) [inline Airport_Ticket_Booking_System_Test/BookingServiceTest.cs]: I do not see test for some methods in BookingService eg cancelBooking

File: Airport_Ticket_Booking_System_Test/FlightServiceTest.cs [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+            var flight1 = new Flight("FL001", "USA", "Canada", "JFK", "YYZ", DateTime.UtcNow, new Dictionary<string, decimal> { { "Economy", 500M } });
+            var flight2 = new Flight("FL002", "UK", "France", "LHR", "CDG", DateTime.UtcNow.AddDays(1), new Dictionary<string, decimal> { { "Economy", 400M } });
+
+            _flightService.AddFlight(flight1); 
+            _flightService.AddFlight(flight2);
+        }
+
+        [Fact]
+        public void GetAllFlights_ShouldReturnAllFlights()
```
Comment (R-abodyak) [inline Airport_Ticket_Booking_System_Test/FlightServiceTest.cs]: you can add third part WhenFlightsExist

Comment (R-abodyak) [reply Airport_Ticket_Booking_System_Test/FlightServiceTest.cs]: you can add another test case when flight list is empty
in this case you may extract the constructor logic in a method and call it when needed

File: Airport_Ticket_Booking_System/Services/CsvService.cs [hunk 7]
```diff
@@ -0,0 +1,162 @@
+using Airport_Ticket_Booking_System.Models;
+
+namespace Airport_Ticket_Booking_System.Services;
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+public static class CsvService
```
Comment (R-abodyak) [inline Airport_Ticket_Booking_System/Services/CsvService.cs]: can you add unit test for this ?

**Review by R-abodyak — CHANGES_REQUESTED** [review]

File: Airport_Ticket_Booking_System/Interfaces/IBookingCsvService.cs [hunk 8]
```diff
@@ -0,0 +1,13 @@
+using Airport_Ticket_Booking_System.Models;
+
+namespace Airport_Ticket_Booking_System.Interfaces;
+
+public interface IBookingCsvService : ICsvService<Booking>
+{
+    void SaveBooking(Booking booking);
+    void UpdateBookingInCsv(Booking updatedBooking);
+    void DeleteBookingFromCsv(string bookingId);
+
```
Comment (R-abodyak) [inline Airport_Ticket_Booking_System/Interfaces/IBookingCsvService.cs:10]: remove extra spaces

File: Airport_Ticket_Booking_System_Test/BookingCsvServiceTest.cs [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+        var updated = new Booking
+        {
+            PassengerName = original.PassengerName,
+            FlightNumber = original.FlightNumber,
+            ClassType = original.ClassType,
+            Price = 150m
+        };
+        original.PassengerName = "Mohammed Sowaity";
+        original.Price = 150;
+        _service.UpdateBookingInCsv(original);
```
Comment (R-abodyak) [inline Airport_Ticket_Booking_System_Test/BookingCsvServiceTest.cs:88]: //act
and // assert are missed

File: Airport_Ticket_Booking_System_Test/BookingServiceTest.cs [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+            new Dictionary<string, decimal> { { "Business", 800M } });
+
+        _bookingService.BookFlight(flight1.FlightNumber, flight1, "Economy");
+        var booking = _bookingService.GetAllBookings().First();
+
+        _bookingService.ModifyBooking(booking.BookingId, flight2, "Business");
+        var updatedBooking = _bookingService.GetAllBookings().First(b => b.BookingId == booking.BookingId);
+
+        Assert.Equal("Business", updatedBooking.ClassType);
+        Assert.Equal("FL002", updatedBooking.FlightNumber);
```
Comment (R-abodyak) [inline Airport_Ticket_Booking_System_Test/BookingServiceTest.cs:57]: use arrange act assert

File: Airport_Ticket_Booking_System_Test/BookingServiceTest.cs [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
```
Comment (R-abodyak) [inline Airport_Ticket_Booking_System_Test/BookingServiceTest.cs:59]: remove extra line

File: Airport_Ticket_Booking_System_Test/BookingCsvServiceTest.cs [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+    {
+        _testFilePath = Path.GetTempFileName();
+        Console.WriteLine("file path " + _testFilePath);
+
+        File.WriteAllText(_testFilePath, "BookingId,PassengerName,FlightNumber,ClassType,Price" + Environment.NewLine);
+
+        _service = new BookingCsvService(_testFilePath);
+    }
+
+    public void Dispose()
```
Comment (R-abodyak) [inline Airport_Ticket_Booking_System_Test/BookingCsvServiceTest.cs:20]: Dispose is not used

File: Airport_Ticket_Booking_System_Test/MenuTests.cs [hunk 13]
```diff
@@ -0,0 +1,78 @@
+// File: Airport_Ticket_Booking_System_Test/MenuTests.cs
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Xunit;
```
Comment (R-abodyak) [inline Airport_Ticket_Booking_System_Test/MenuTests.cs:5]: remove unused namespaces

File: Airport_Ticket_Booking_System_Test/MenuTests.cs [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+            _fixture = new Fixture().Customize(new AutoMoqCustomization());
+            _flightServiceMock = _fixture.Freeze<Mock<IFlightService>>();
+            _bookingServiceMock = _fixture.Freeze<Mock<IBookingService>>();
+
+            var manager = new Manager(_flightServiceMock.Object, _bookingServiceMock.Object);
+            _menu = new Menu(manager);
+        }
+
+        [Fact]
+        public void BookFlight_ShouldBookFlight_WhenFlightExists()
```
Comment (R-abodyak) [inline Airport_Ticket_Booking_System_Test/MenuTests.cs:34]: keep going with the 3 section naming 
It's clear!

File: Airport_Ticket_Booking_System_Test/BookingCsvServiceTest.cs [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+        });
+
+
+        var updated = new Booking
+        {
+            PassengerName = original.PassengerName,
+            FlightNumber = original.FlightNumber,
+            ClassType = original.ClassType,
+            Price = 150m
+        };
```
Comment (R-abodyak) [inline Airport_Ticket_Booking_System_Test/BookingCsvServiceTest.cs:85]: unused
